public/js/register.js

```diff
@@ -0,0 +1,82 @@
+import {
+    inputEnabled,
+    setDiv,
+    message,
+    token,
+    enableInput,
+    setToken,
+} from "./index.js";
+import { showLoginRegister } from "./loginRegister.js";
+import { showTasks } from "./tasks.js";
+  
+let registerDiv = null;
+let name = null;
+let email1 = null;
+let password1 = null;
+let password2 = null;
+  
+export const handleRegister = () => {
+    registerDiv = document.getElementById("register-div");
+    name = document.getElementById("name");
+    email1 = document.getElementById("email1");
+    password1 = document.getElementById("password1");
+    password2 = document.getElementById("password2");
+    const registerButton = document.getElementById("register-button");
+    const registerCancel = document.getElementById("register-cancel");
+  
+    registerDiv.addEventListener("click", async (e) => {
+        if (inputEnabled && e.target.nodeName === "BUTTON") {
+            if (e.target === registerButton) {
+                if(password1.value != password2.value) {
+                    message.textContent = "The passwords don't match";
+                }
+                else {
+                    enableInput(false);
+                    try {
+                        const response = await fetch("/api/auth", {
```

Comment:
The correct route for register is `/api/auth/route`. It doesn't work the way it is right now.